server.go

```diff
@@ -2250,7 +2251,10 @@ func (s *server) peerDoneHandler(sp *serverPeer) {
 
 	// Notify the net sync manager the peer is gone if it was ever notified that
 	// the peer existed.
-	if sp.syncNotified {
+	sp.syncNotifiedMtx.Lock()
+	syncNotified := sp.syncNotified
+	sp.syncNotifiedMtx.Unlock()
```

Comment:
Could just be a read lock, or an atomic.

Reply:
Could be a read lock yes, but a read/write lock should work just fine as well.

Reply:
Ok!

Reply:
There is really no reason to use a read lock or atomic here because the lock is only taken once when a peer connects and once it disconnects which are both one time events.  If it were in a hot path, cutting down lock contention would indeed be something watch out for.